### DISCUSSION TITLE
Is there an auto_route oriented alternative to directly calling 'showDialog'?

### DISCUSSION THREAD

**Discussion by Adam-Langley** [Milad-Akarie/auto_route_library#1712] [Q&A]
It feels odd to navigate throughout the application using the Router, but resort to 'showDialog' when I want to display a modal settings screen.

Is there a way to overlay an '@RoutePage' over the existing application without interrupting it? Kind of like co-existing routes, one of them is paused while the other overlays it?

Or some better way to do it that has perhaps become the defacto way of doing it with auto_route, but just not been documented yet?

Thanks!

**Comment by Adam-Langley** [comment]
I think 'fullscreenDialog' on the route is the way to do it...

**Reply by Adam-Langley** [reply]
Maybe I closed this too soon... by using the fullscreenDialog parameters it blanks out the existing route... so it's not working as an overlay

**Comment by Shreemanarjun** [comment]
You can use like this in route declaration.


```dart
    CustomRoute(
      page: ConfirmConnectionDialogRoute.page,
      path: '/confirmation',
      customRouteBuilder:
          <T>(BuildContext context, Widget child, AutoRoutePage<T> page) {
        return DialogRoute(
          context: context,
          // this is important
          settings: page,
          builder: (_) => child,
        );
      },
    ),

```

**Reply by Adam-Langley** [reply]
Thanks @Shreemanarjun 
With your sample code, I still see the same results - the main window turns black as the dialog is overlayed...

<img width="1274" alt="image" src="https://github.com/Milad-Akarie/auto_route_library/assets/16373742/f296908d-3446-4474-b6f2-12687ab86d73">
